### PR TITLE
Fix Find similar domains nav item for domain only sites

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* For domain only sites the "Find similar domains" nav item wasn't working because there is no route to add new domain on the same site. We should open the domain-only flow when you click on this button

#### Testing instructions

* Open a domain only status page and click on the "Find similar domains" nav item - it should open the domain-only flow with the search term populated

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx [hunk 1]
```diff
@@ -392,9 +393,20 @@ class DomainManagementNavigationEnhanced extends React.Component {
 		// we don't use the full domain name, to avoid an error about the taken domain
 		const searchTerm = domain.name.split( '.' )[ 0 ];
 
+		let path;
+
+		if ( selectedSite && selectedSite.options && selectedSite.options.is_domain_only ) {
```
Comment (stephanethomas) [inline client/my-sites/domains/domain-management/edit/navigation/enhanced.jsx:398]: I think we should be able to simplify this condition with:

```javascript
if ( selectedSite?.options?.is_domain_only ) {
```

**Comment by nsakaimbo** [comment]
Is this PR branch rebased on latest Calypso master? (Earliest commit is from 6 days ago so there's a chance you don't have the fix that was merged for wp-desktop yesterday.)

**Comment by hambai** [comment]
> Is this PR branch rebased on latest Calypso master? (Earliest commit is from 6 days ago so there's a chance you don't have the fix that was merged for wp-desktop yesterday.)

I just rebased it

**Comment by nsakaimbo** [comment]
Looks like the `ci/wp-desktop` job doesn't run if the PR is already approved. Re-ran in a separate test branch and CI for the rebased branch passes ([link](https://app.circleci.com/pipelines/github/Automattic/wp-desktop/15763/workflows/63fd6f97-a029-46a5-97ec-857f6b34a363/jobs/64262)).

**Pull request merged**